app/src/main/java/org/worldbank/transport/driver/utilities/RecordFormPaginator.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    public static String getPluralTitle(Field sectionField, String defaultTitle) {
+
+        if (sectionField == null) {
+            Log.e(LOG_LABEL, "Cannot find plural label for unknown field");
+            return defaultTitle;
+        }
+
+        PluralTitle pluralAnnotation = sectionField.getAnnotation(PluralTitle.class);
+        if (pluralAnnotation != null) {
+            String pluralTitle = pluralAnnotation.value();
+            if (pluralTitle.length() > 0) {
+                return pluralTitle;
+            } else {
+                Log.w(LOG_LABEL, "No plural title found for section");
+            }
+        } else {
+            Log.w(LOG_LABEL, "No plural title found for section");
+        }
+
+        return defaultTitle;
+    }
+
+    public static String getSingleTitle(Field sectionField, String defaultTitle) {
+        Title titleAnnotation = sectionField.getAnnotation(Title.class);
+        if (titleAnnotation != null) {
+            String title = titleAnnotation.value();
+            if (title.length() > 0) {
+                return title;
+            } else {
+                Log.w(LOG_LABEL, "No title found for section");
+            }
+        } else {
+            Log.w(LOG_LABEL, "No title found for section");
+        }
+
+        return defaultTitle;
+    }
+
+    public static Class getSectionClass(String sectionName) {
+        try {
+            return Class.forName(MODEL_PACKAGE + sectionName);
+        } catch (ClassNotFoundException e) {
+            Log.e(LOG_LABEL, "Could not fine class named " + sectionName);
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the currently editing object for a given section, or create it if adding a record.
+     *
+     * @param sectionField Field for the section on the DriverSchema class
+     * @param sectionClass Class of the field
+     * @param currentlyEditing DriverSchema object currently editing, as managed by the app singleton
+     * @return Section object, ready for use by form
+     */
+    public static Object getOrCreateSectionObject(Field sectionField, Class sectionClass, Object currentlyEditing) {
+        // attempt to get the section from the currently editing model object;
+        // it will not exist if creating a new record
+
+        try {
+            Object section = sectionField.get(currentlyEditing);
+
+            if (section == null) {
+                Log.d(LOG_LABEL, "No section found for field " + sectionField.getName());
+                // instantiate a new thing then
+                section = sectionClass.newInstance();
+
+                // add the new section to the currently editing model object
+                sectionField.set(currentlyEditing, section);
+
+                if (sectionField.get(currentlyEditing) == null) {
+                    Log.e(LOG_LABEL, "Section field is still null after set to new instance!");
```

Comment:
This case will return null, as do the other error cases. There's a check for a null section in the form build, which will return early after also logging a message, resulting in a blank view and a hopefully more readable log trace.

I've generally added handlers with error logs instead of throwing things because the code would otherwise get cluttered quickly with try/catch clauses for all the things that can go wrong with introspection.

This particular case has never happened to me.